### ISSUE TITLE
List/contains definition unclear

### ISSUE THREAD

**Issue by littledan** [whatwg/infra#230]
From https://infra.spec.whatwg.org/#list-contain:

> A list contains an item if it appears in the list. We can also denote this by saying that, for a list list and an index index, "list[index] exists". 

I'm confused about a couple points here:
- Is the equality algorithm somehow implied by "appears", or is it described somewhere else?
- How does the second sentence relate to the first?

(Context: I'm wondering about how to use an Infra set for the `[[PlatformBrand]]` (https://github.com/heycam/webidl/issues/97), and possibly also for private method brands (https://github.com/tc39/proposal-decorators/issues/180). It looks like I'd use this `contains` algorithm to check membership, right?)

**Comment by annevk** [comment]
See also #106. I think we mostly want object identity here, except for primitive data types. Most specifications already assume those semantics when it comes to equality.

I think the second sentence doesn't really relate to the first that much and probably needs to be reworded. Maybe something like:

> For a list _list_ and an index _index_, _list_[_index_] exists, if _index_ is not out-of-bound for _list_.

**Comment by annevk** [comment]
(I'd expect `[[PlatformBrand]]` to be a set, but yeah, you'd use contains.)

**Comment by littledan** [comment]
Can sets contain things like Records? I'm not sure if those are considered to have object identity.

So the second sentence was just an idependent definition of "exists"?

**Comment by littledan** [comment]
2. Is "appears" interpreted to reference the local definition of "exists"? It's unclear to me how.

**Comment by annevk** [comment]
No, I think it's meant as looping over the list, comparing the input item with the items that result from looping, and returning true if they're "equal". Anyway, I agree with you that this could use some refactoring.

**Comment by domenic** [comment]
Some definitions in Infra are fairly vacuous. E.g. contains delegates to an undefined "appears in" concept, iterate to "on each item in order", append and prepend to "add". The intention is roughly "you already know what this vague concept is, but here's the exact word you can use and reference".

We could probably redefine "contains" in terms of equals and iterate, instead of "appears". But I'm not sure how far we should pursue such a program. The endpoint of making these definitions rigorous is basically axiomatic set theory.
